File: Agent14_Phase15_TestingQA_Completed/agent14-academic-risk-phase2/backend/app/schemas/institutional_ai.py
from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class InstitutionalAIIntent(str, Enum):
    EXECUTIVE_SUMMARY = "executive_summary"
    RISK_ANALYSIS = "risk_analysis"
    MENTOR_WORKLOAD = "mentor_workload"
    INTERVENTION_COVERAGE = "intervention_coverage"
    PRIORITY_REVIEW = "priority_review"


class InstitutionalAIRequest(BaseModel):
    intent: InstitutionalAIIntent = InstitutionalAIIntent.EXECUTIVE_SUMMARY
    focus: Optional[str] = Field(default=None, max_length=300)


class InstitutionalAIResponse(BaseModel):
    role: str
    scope: str
    intent: InstitutionalAIIntent
    provider: str
    model: str
    grounded: bool
    title: str
    executive_summary: str
    key_findings: list[str]
    recommended_actions: list[str]
    cautions: list[str]
    source_snapshot: dict

File: Agent14_Phase15_TestingQA_Completed/agent14-academic-risk-phase2/frontend/src/features/institutionalAI.ts
import api from "../services/api";

export type InstitutionalAIIntent =
  | "executive_summary"
  | "risk_analysis"
  | "mentor_workload"
  | "intervention_coverage"
  | "priority_review";

export interface InstitutionalAIResponse {
  role: "hod" | "dean";
  scope: string;
  intent: InstitutionalAIIntent;
  provider: string;
  model: string;
  grounded: boolean;
  title: string;
  executive_summary: string;
  key_findings: string[];
  recommended_actions: string[];
  cautions: string[];
  source_snapshot: {
    risk_source: string;
    risk_types: string[];
    scope: Record<string, string>;
  };
}

export async function runInstitutionalAI(intent: InstitutionalAIIntent, focus?: string) {
  const response = await api.post<InstitutionalAIResponse>("/institutional-ai/analyze", {
    intent,
    focus: focus?.trim() || undefined,
  });
  return response.data;
}
